### PR TITLE
feat(Buttons): Leading + trailing icons

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Adds standardized leading and trailing icons to our Fill, Stroke, and TextButtons.

Also reorged some of our buttons files

### PR Checklist

- [x] Related to designs: 
- [x] Related to JIRA ticket: [GM-548]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

Check out the Inline Icons section of the PR env ([here](https://65c5389979908a0eceec4bdd--gamut-preview.netlify.app/?path=/docs/atoms-button--cta-button#inline-icon)) and the examples in the PRs linked below. 

image

#### PR Links and Envs

I am still writing testing instructions for the monorepo, but you will see CI passing there

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37071)  | [TBD](http://www.google.fr/ 'Named link title') |
| Monorepo       | [Monorepo Link](https://github.com/codecademy-engineering/mono/pull/4881')  | [TBD](http://www.google.fr/ 'Named link title')   |

[GM-548]: https://codecademy.atlassian.net/browse/GM-548?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c12bd45ee042075329f80d--gamut-preview.netlify.app>https://65c12bd45ee042075329f80d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c12bd45ee042075329f80d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c13cb0cef48a0ceb04066f--gamut-preview.netlify.app>https://65c13cb0cef48a0ceb04066f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c13cb0cef48a0ceb04066f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.1-alpha.7efa8a.0<br/>@codecademy/gamut-kit@0.6.378-alpha.7efa8a.0<br/>@codecademy/styleguide@66.10.1-alpha.7efa8a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c1522537d38723d3bb2fc2--gamut-preview.netlify.app>https://65c1522537d38723d3bb2fc2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c1522537d38723d3bb2fc2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.1-alpha.367953.0<br/>@codecademy/gamut-kit@0.6.378-alpha.367953.0<br/>@codecademy/styleguide@66.10.1-alpha.367953.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c158f69ab25726cc14047e--gamut-preview.netlify.app>https://65c158f69ab25726cc14047e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c158f69ab25726cc14047e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.3-alpha.c6d744.0<br/>@codecademy/gamut-kit@0.6.380-alpha.c6d744.0<br/>@codecademy/styleguide@66.10.3-alpha.c6d744.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c3f4e2c45fba02d4d951b4--gamut-preview.netlify.app>https://65c3f4e2c45fba02d4d951b4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c3f4e2c45fba02d4d951b4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.fafedf.0<br/>@codecademy/gamut-kit@0.6.381-alpha.fafedf.0<br/>@codecademy/styleguide@66.10.4-alpha.fafedf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c5389979908a0eceec4bdd--gamut-preview.netlify.app>https://65c5389979908a0eceec4bdd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c5389979908a0eceec4bdd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c541da3eab3416981d10ef--gamut-preview.netlify.app>https://65c541da3eab3416981d10ef--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c541da3eab3416981d10ef>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.bef8c5.0<br/>@codecademy/gamut-kit@0.6.381-alpha.bef8c5.0<br/>@codecademy/styleguide@66.10.4-alpha.bef8c5.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.919f4c.0<br/>@codecademy/gamut-kit@0.6.381-alpha.919f4c.0<br/>@codecademy/styleguide@66.10.4-alpha.919f4c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c64af08e0a4410dc898e37--gamut-preview.netlify.app>https://65c64af08e0a4410dc898e37--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c64af08e0a4410dc898e37>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.dd7e0a.0<br/>@codecademy/gamut-kit@0.6.381-alpha.dd7e0a.0<br/>@codecademy/styleguide@66.10.4-alpha.dd7e0a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65c667de972428234517c3b9--gamut-preview.netlify.app>https://65c667de972428234517c3b9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65c667de972428234517c3b9>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]
lgtm!

File: packages/gamut/src/Button/StrokeButton.tsx [hunk 1]
```diff
@@ -1,7 +1,64 @@
...
+//         aria-label={7}
+//         icon={MiniKebabMenuIcon}
+//         size="never"
+//         tabIndex={-1}
+//         variant="65"
+//         mb={4}
+//       />
+//     </>
+//   );
+// };
```
Comment (jakemhiller) [inline packages/gamut/src/Button/StrokeButton.tsx:64]: obligatory "remove this before merging" comment

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.7db30e.0<br/>@codecademy/gamut-kit@0.6.381-alpha.7db30e.0<br/>@codecademy/styleguide@66.10.4-alpha.7db30e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65cd294aad5f4b27da153be3--gamut-preview.netlify.app>https://65cd294aad5f4b27da153be3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65cd294aad5f4b27da153be3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.4b36ef.0<br/>@codecademy/gamut-kit@0.6.381-alpha.4b36ef.0<br/>@codecademy/styleguide@66.10.4-alpha.4b36ef.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.12.4-alpha.fde610.0<br/>@codecademy/gamut-kit@0.6.381-alpha.fde610.0<br/>@codecademy/styleguide@66.10.4-alpha.fde610.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://65cd373cd2a5fb303914a7ca--gamut-preview.netlify.app>https://65cd373cd2a5fb303914a7ca--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/65cd373cd2a5fb303914a7ca>Deploy Logs</a></p>

**Pull request merged**